app/src/store/store.ts

```diff
@@ -157,11 +159,18 @@ export class Store {
    * makes the initial API calls to fetch the data we need to display in the app
    */
   async fetchAllData() {
+    await this.subServerStore.fetchStatus();
     await this.nodeStore.fetchInfo();
     await this.channelStore.fetchChannels();
-    await this.swapStore.fetchSwaps();
     await this.nodeStore.fetchBalances();
     await this.sessionStore.fetchSessions();
+
+    if (
+      this.subServerStore.subServers.loop?.running &&
+      !this.subServerStore.subServers.loop?.error
+    ) {
+      await this.swapStore.fetchSwaps();
```

Comment:
If our subserver loop state changes, will the store be refreshed ?

Reply:
@jbrill After talking with Elle she explained that the subserver status server does not update when runtime errors are encountered. Currently the subserver status server only catches errors on startup.

Since we aren't polling the subserver status server for changes, there's no way for the state to change in the background. I should make sure that we fetch updated subserver information when navigating between pages though.

Reply:
yeah - currently we can only check if it started successfully and there isnt currently a way to ping the server to see if it has crashed after startup (at least not at the moment)

Reply:
Gotcha -- thanks for the explanation 👍